src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -189,20 +205,28 @@ public function populate( $thread_id = 0, $order = 'ASC', $args = array() ) {
 			return false;
 		}
 
-		$last_message_index         = count( $this->messages ) - 1;
-		$this->last_message_id      = $this->messages[ $last_message_index ]->id;
-		$this->last_message_date    = $this->messages[ $last_message_index ]->date_sent;
-		$this->last_sender_id       = $this->messages[ $last_message_index ]->sender_id;
-		$this->last_message_subject = $this->messages[ $last_message_index ]->subject;
-		$this->last_message_content = $this->messages[ $last_message_index ]->message;
+		// Messages count.
+		$this->messages_count = self::get_total_thread_message_count( $this->thread_id );
 
-		foreach ( (array) $this->messages as $key => $message ) {
-			$this->sender_ids[ $message->sender_id ] = $message->sender_id;
+		// Sender IDs.
+		$this->sender_ids = self::get_thread_sender_ids( $this->thread_id );
+
+		// Get and set last message.
+		$message = self::get_latest_thread_message( $this->thread_id );
```

Comment:
If I understand well this part: you suggest to add a new query to get the latest thread's message instead of finding the latest thread's message that should be into the `$this->messages`? 
Is this a downsize of introducing messages' thread pagination?
Could we avoid this new query, making sure to include the latest thread's message ID, for example ?

Reply:
Not related to the pagination. This is related to the default behavior of requesting ALL messages by default. Which is not very performant for apps with thousands of messages.